api/converter_utils.go

```diff
@@ -407,6 +408,101 @@ func signedTxnWithAdToTransaction(stxn *transactions.SignedTxnWithAD, extra rowD
 		}
 
 		application = &a
+	case protocol.StateProofTx:
+		partPath := make([][]byte, len(stxn.Txn.StateProof.PartProofs.Path))
+		for idx, part := range stxn.Txn.StateProof.PartProofs.Path {
+			digest := make([]byte, len(part))
+			copy(digest, part)
+			partPath[idx] = digest
+		}
+
+		sigProofPath := make([][]byte, len(stxn.Txn.StateProof.SigProofs.Path))
+		for idx, sigPart := range stxn.Txn.StateProof.SigProofs.Path {
+			digest := make([]byte, len(sigPart))
+			copy(digest, sigPart)
+			sigProofPath[idx] = digest
+		}
+
+		// We need to iterate through these in order, to make sure our responses are deterministic
+		keys := make([]uint64, len(stxn.Txn.StateProof.Reveals))
+		elems := 0
+		for key := range stxn.Txn.StateProof.Reveals {
+			keys[elems] = key
+			elems++
+		}
+		sort.Slice(keys, func(i, j int) bool { return keys[i] < keys[j] })
+		reveals := make([]generated.StateProofReveal, len(stxn.Txn.StateProof.Reveals))
+		for _, key := range keys {
+			revToConv := stxn.Txn.StateProof.Reveals[key]
+			commitment := revToConv.Part.PK.Commitment[:]
+			falconSig := []byte(revToConv.SigSlot.Sig.Signature)
+			verifyKey := revToConv.SigSlot.Sig.VerifyingKey.PublicKey[:]
+			proofPath := make([][]byte, len(revToConv.SigSlot.Sig.Proof.Path))
+			for idx, proofPart := range revToConv.SigSlot.Sig.Proof.Path {
+				proofPath[idx] = proofPart
+			}
+
+			reveals[key] = generated.StateProofReveal{
+				Participant: &generated.StateProofParticipant{
+					Verifier: &generated.StateProofVerifier{
+						Commitment:  &commitment,
+						KeyLifetime: uint64Ptr(revToConv.Part.PK.KeyLifetime),
+					},
+					Weight: uint64Ptr(revToConv.Part.Weight),
+				},
+				Position: uint64Ptr(key),
+				SigSlot: &generated.StateProofSigSlot{
+					LowerSigWeight: uint64Ptr(revToConv.SigSlot.L),
+					Signature: &generated.StateProofSignature{
+						FalconSignature:  &falconSig,
+						MerkleArrayIndex: uint64Ptr(revToConv.SigSlot.Sig.VectorCommitmentIndex),
+						Proof: &generated.MerkleArrayProof{
+							HashFactory: &generated.HashFactory{
+								HashType: uint64Ptr(uint64(revToConv.SigSlot.Sig.Proof.HashFactory.HashType)),
+							},
+							Path:      &proofPath,
+							TreeDepth: uint64Ptr(uint64(revToConv.SigSlot.Sig.Proof.TreeDepth)),
+						},
+						VerifyingKey: &verifyKey,
+					},
+				},
+			}
+		}
+		proof := generated.StateProof{
```

Comment:
I think we should also copy `PositionsToReveal` slice

Reply:
@Eric-Warehime You can use `byteSlicePtr` to create a copy